integration_tests/incentives_test.go

```diff
@@ -106,24 +138,66 @@ func (s *IntegrationTestSuite) TestIncentives() {
 		s.T().Log("proposal approved!")
 
 		s.T().Log("verifying parameter was changed")
-		incentivesParamsRes, err := incentivesQueryClient.QueryParams(ctx, &incentivestypes.QueryParamsRequest{})
+		incentivesParamsRes, err = incentivesQueryClient.QueryParams(ctx, &incentivestypes.QueryParamsRequest{})
 		s.Require().NoError(err)
 		s.Require().Equal(incentivesParamsRes.Params.IncentivesCutoffHeight, uint64(1000))
 
-		s.T().Log("verifying that the base distribution rate is greater than zero")
-		beforeAmount = s.queryDelegationRewardAmount(ctx, val.keyInfo.GetAddress().String(), distQueryClient)
+		s.T().Log("verifying that the base distribution rate is the expected value")
+		beforeAmount, beforeHeight := s.getRewardAmountAndHeight(ctx, distQueryClient, valOperatorAddress, clientCtx)
 		time.Sleep(time.Second * 12)
-		afterAmount = s.queryDelegationRewardAmount(ctx, val.keyInfo.GetAddress().String(), distQueryClient)
-		s.T().Logf("before: %s, after: %s", beforeAmount.String(), afterAmount.String())
+		afterAmount, afterHeight := s.getRewardAmountAndHeight(ctx, distQueryClient, valOperatorAddress, clientCtx)
+		// Assumes each validator has equally weight bonding power
+		actualDistributionPerBlock := (afterAmount.Sub(beforeAmount)).Quo(sdk.NewDec(afterHeight - beforeHeight)).Mul(sdk.NewDec(int64(len(s.chain.validators))))
+		s.T().Logf("before: %s, after: %s, blocks %d-%d", beforeAmount.String(), afterAmount.String(), beforeHeight, afterHeight)
 		s.Require().True(afterAmount.GT(beforeAmount))
+		s.Require().Equal(expectedDistributionPerBlock.Amount.ToDec(), actualDistributionPerBlock)
```

Comment:
Should we also be checking that the cutoff height is working appropriately? We'd either have to set a value less than 1000 in the prop that we'd wait for, or execute another prop to lower the value below the current height and observe no more emissions.

Reply:
Bumping this comment as it's the only other thing left unaddressed. What are your thoughts on this?